signature/signature.go

```diff
@@ -27,37 +27,70 @@ func (err InvalidSignatureError) Error() string {
 }
 
 // Signature is a parsed content of a signature.
+// The only way to get this structure from a blob should be as a return value from a successful call to verifyAndExtractSignature below.
 type Signature struct {
 	DockerManifestDigest digest.Digest
 	DockerReference      string // FIXME: more precise type?
 }
 
-// Wrap signature to add to it some methods which we don't want to make public.
-type privateSignature struct {
-	Signature
+// untrustedSignature is a parsed content of a signature.
+type untrustedSignature struct {
+	UntrustedDockerManifestDigest digest.Digest
+	UntrustedDockerReference      string // FIXME: more precise type?
+	UntrustedCreatorID            *string
+	UntrustedTimestamp            *int64
```

Comment:
Right now the only consumer is `GetUntrustedSignatureInformationWithoutVerifying`, so placement of the conversion makes no difference to the total complexity of the package.

Using the native `int64` here is nice in that it explicitly corresponds to the JSON semantics, and we are more certain that we can round-trip the value and that the caller setting the value is aware about the conversion (e.g., compared to `time.Time`, that this records neither timezones nor sub-second resolutions).

Though, perhaps, before we start using this field in earnest and before we ship the first version of a validating parser of this fields, perhaps we should change the type to a `float64` (with its sub-second resolution and a more restricted precision), which is what JSON _actually_ stores? The `int64` also has conversion issues. @runcom  ?